src/main/java/com/example/auctionapp/repository/CategoryRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.example.auctionapp.repository;
+
+import com.example.auctionapp.entity.Category;
+import org.springframework.data.domain.Page;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import org.springframework.data.domain.Pageable;
+import java.util.List;
+
+public interface CategoryRepository extends JpaRepository<Category, Long> {
+
+    Page<Category> findAllByName(String name, Pageable pageable);
```

Comment:
Does category need fetching by page, category isn't big object. All of them can be fetched at once as a list